src/irmin/object_graph.ml

```diff
@@ -31,26 +31,34 @@ let list_partition_map f t =
   in
   aux [] [] t
 
-module Make (Hash : HASH) (Branch : Type.S) = struct
+module Make
+    (Contents : Type.S)
+    (Node : Type.S)
+    (Commit : Type.S)
+    (Branch : Type.S) =
+struct
   module X = struct
     type t =
-      [ `Contents of Hash.t
-      | `Node of Hash.t
-      | `Commit of Hash.t
+      [ `Contents of Contents.t
+      | `Node of Node.t
+      | `Commit of Commit.t
       | `Branch of Branch.t ]
     [@@deriving irmin]
```

Comment:
`Contents` -> `Contents_id`, `Node` -> `Node_id`, ...?

At first look it seems like this piece of code manipulates objects and not ids.

Reply:
Agreed. I think I prefer `key` as a name for this concept vs. `id`, so will try that.